airflow/executors/celery_executor.py

```diff
@@ -262,10 +268,26 @@ def _process_tasks(self, task_tuples_to_send: List[TaskInstanceInCelery]) -> Non
         self.log.debug('Sent all tasks.')
 
         for key, _, result in key_and_async_results:
+            if isinstance(result, ExceptionWithTraceback) and isinstance(
+                result.exception, AirflowTaskTimeout
+            ):
+                if key in self.task_publish_retries and (
+                    self.task_publish_retries.get(key) <= self.task_publish_max_retries
+                ):
+                    Stats.incr("celery.task_timeout_error")
```

Comment:
Should we add this metric to https://airflow.readthedocs.io/en/latest/logging-monitoring/metrics.html?

Reply:
https://github.com/apache/airflow/pull/12152